backend/rates/coingecko.go

```diff
@@ -0,0 +1,153 @@
+package rates
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"sort"
+	"strconv"
+	"time"
+
+	"github.com/digitalbitbox/bitbox-wallet-app/util/observable"
+	"github.com/digitalbitbox/bitbox-wallet-app/util/observable/action"
+)
+
+const coingeckoAPIV3 = "https://api.coingecko.com/api/v3"
+
+// forwardHistoryLoop periodically updates historical market exchange rates
+// forward in time starting with the last fetched timestamp.
+func (updater *RateUpdater) forwardHistoryLoop() {
+	for {
+		// When to update next, after this loop iteration is done.
+		untilNext := 10 * time.Minute // TODO: add jitter
+
+		start := updater.histLatestTimestamp("BTC")
+		if !start.IsZero() {
+			// Start slightly past the last fetched timestamp
+			start = start.Add(10 * time.Minute)
+			now := time.Now()
+			// It's ok if start == now or now < start. CoinGecko simply returns no results.
+			// It's also ok if our "now" doesn't match CoinGecko: it always returns
+			// the latest available data without errors.
+			err := updater.updateHist("BTC", "USD", start, now)
```

Comment:
"BTC" is used as the coin id in the coingecko api, but BTC doesn't exist, it is "bitcoin":

https://www.coingecko.com/api/documentations/v3#/coins/get_coins_list

At least if I enter `BTC` here it does not work:

https://www.coingecko.com/api/documentations/v3#/coins/get_coins__id__market_chart_range

Reply:
yeah, coingecko uses IDs and `bitcoin` is their identifier for BTC. you'll see it in `fetchMarkerRange`. this will need a map between the standard symbols to coingecko API.

Reply:
actually, forget what I said. I made it too generic and forgot about the mapping :see_no_evil:. it is currently fetching no useful data indeed.

Reply:
done